openfeature/telemetry.go

```diff
@@ -0,0 +1,95 @@
+package openfeature
+
+import (
+	"strings"
+)
+
+type EvaluationEvent struct {
+	Name       string
+	Attributes map[string]interface{}
+	Body       map[string]interface{}
+}
+
+const (
+	// The OpenTelemetry compliant event attributes for flag evaluation.
+	// Specification: https://opentelemetry.io/docs/specs/semconv/feature-flags/feature-flags-logs/
+
+	TelemetryKey string = "feature_flag.key"
+	TelemetryErrorCode string = "error.type"
+	TelemetryVariant string = "feature_flag.variant"
+	TelemetryContextID string = "feature_flag.context.id"
+	TelemetryErrorMsg string = "feature_flag.evaluation.error.message"
+	TelemetryReason string = "feature_flag.evaluation.reason"
+	TelemetryProvider string = "feature_flag.provider_name"
+	TelemetryFlagSetID string = "feature_flag.set.id"
+	TelemetryVersion string = "feature_flag.version"
+
+
+	// Well-known flag metadata attributes for telemetry events.
+	// Specification: https://openfeature.dev/specification/appendix-d#flag-metadata
+	TelemetryFlagMetaContextId string = "contextId"
+	TelemetryFlagMetaFlagSetId string = "flagSetId"
+	TelemetryFlagMetaVersion string = "version"
+
+	// OpenTelemetry event body.
+	// Specification: https://opentelemetry.io/docs/specs/semconv/feature-flags/feature-flags-logs/
+	TelemetryBody string = "value"
+
+	FlagEvaluationEventName string = "feature_flag.evaluation"
+)
+
+func CreateEvaluationEvent(hookContext HookContext, details InterfaceEvaluationDetails) EvaluationEvent {
```

Comment:
@beeme1mr @chrfwow @lukas-reining 

I have been looking over this again and using the `InterfaceEvalutionDetails` works but doesn't keep strong types so I was searching for input on if there should be some refactoring to have 2 `CreateEvaluationEvent` one for the stricter types using generics and one for Objects. 

I know what is here is working for test but I wanted to ask.

```suggestion
type TelemetryEvaluationValueType interface {
	~bool | ~string | ~float64 | ~int64 
}

type TelemetryEvaluationDetails[T TelemetryEvaluationValueType] struct {
	FlagKey string
	FlagType Type
	Value T
	ResolutionDetail
}

func CreateEvaluationEvent[T TelemetryEvaluationValueType](hookContext HookContext, details TelemetryEvaluationDetails[T]) EvaluationEvent {}

func CreateObjectFlagEvaluationEvent(hookContext HookContext, details InterfaceEvaluationDetails) EvaluationEvent {}
```

Reply:
Sounds, good! The more I looked the more I was like 🤔 haha. Happy it got rereviewed even just to answer this question.